cranelift/codegen/src/isa/x64/lower/isle.rs

```diff
@@ -44,6 +44,7 @@ use target_lexicon::Triple;
 type BoxCallInfo = Box<CallInfo>;
 type BoxVecMachLabel = Box<SmallVec<[MachLabel; 4]>>;
 type MachLabelSlice = [MachLabel];
+type BoxArgInfo = Box<ArgInfo>;
```

Comment:
If you move the vectors outside `ArgInfo`, making it `Vec<ArgInfo>` or similar, then `BoxArgInfo` could be defined as `Box<[ArgInfo]>` instead of boxing the vector. Since this would be a fat pointer, the box needs space for two `usize` values instead of one, but I'm guessing that wouldn't make any of the backends' `Inst` representations grow. I assume when you decided to box this that you were really trying to avoid sticking two `Vec`s in the instructions, which would need _six_ `usize` values.

Reply:
Updated! (Actually as a `Vec<ArgPair>` for now as the extra usize doesn't make a difference in the overall inst enum size and it keeps things a bit simpler otherwise)

And indeed, the original out-of-lining of the data was because otherwise the `Inst` types all became larger; but no longer.